src/soil_moisture_profile.cxx

```diff
@@ -912,38 +915,38 @@ Reads 1D data from the config file
 */
 
 vector<double> soil_moisture_profile::
-ReadVectorData(string key)
+ReadVectorData(string param_name, string param_value)
 {
   int pos =0;
   string delimiter = ",";
-  vector<double> value(0.0);
-  string z1 = key;
+  vector<double> values(0.0);
+  string z1 = param_value;
   
   if (z1.find(delimiter) == string::npos) {
     double v = stod(z1);
     if (v == 0.0) {
```

Comment:
If it makes sense, I would suggest slightly altering the conditional to the following: 
```suggestion
    if (v <= 0.0) {
```